app/models/api_docs/service.rb

```diff
@@ -1,32 +1,38 @@
 # encoding: utf-8
+
 class ApiDocs::Service < ApplicationRecord
   CURRENT_SWAGGER_VERSION = 2
 
   include SystemName
   extend System::Database::Scopes::IdOrSystemName
 
   belongs_to :account, required: true
-  belongs_to :service, class_name: '::Service', inverse_of: :api_docs_services
+  belongs_to :service, class_name: '::Service', inverse_of: :deprecated_api_docs, required: false
+  belongs_to :owner, polymorphic: true, required: false
 
   attr_accessible :account, :body, :name, :description, :published, :skip_swagger_validations
   attr_readonly :system_name
 
-  self.table_name = "api_docs_services"
+  self.table_name = 'api_docs_services'
   has_system_name uniqueness_scope: :account_id
 
   include ServiceDiscovery::ModelExtensions::ApiDocs::Service
 
   validates :name, :body, presence: true
   validates :name, :system_name, :base_path, :swagger_version, length: { maximum: 255 }
+  validates :owner_type, length: { maximum: 255, allow_blank: true }
   validates :description, length: { maximum: 65535 }
   validates :body, length: { maximum: 4294967295, allow_blank: true }
-  validate :service_belongs_to_account, if: -> { service_id.present? && service_id_changed? }
+  validate :service_belongs_to_account, if: -> { service_id.present? || owner_type == 'Service' }
 
-  before_validation(on: :create) { self.account ||= service&.account }
+  before_validation(on: :create) { self.account ||= owner_account }
+  before_validation(on: :save) { self.owner ||= account }
```

Comment:
This will mandate a migration to pre-fill `owner_type` and `owner_id` of all service-owned api_docs before we have any process running this version of the code. Perhaps we can make this a bit smoother... Well, maybe not needed. Related to https://github.com/3scale/porta/pull/2116/files?file-filters%5B%5D=.rb#r489444922.